### DISCUSSION TITLE
How to populate postgis db with osm data?

### DISCUSSION THREAD

**Discussion by berendjan** [maplibre/martin#1138] [Q&A]
The readme does not show how to populate a postgis db properly such that it usable by martin. It it possible to add this, or a link to somewhere if it’s explained there. Can i use osm2pgsql for example?

**Comment by nyurik** [comment]
Any table that has a geometry column can be exposed by martin - so yes, anything generated by osm2pgsql can be published already. In the next release (already on the main branch), martin has improved how it detects tile-returning functions too, so the easiest path would be to use openmaptiles project to generate the mvt function.  Any documentation PRs are welcome :)

**Comment by berendjan** [comment]
Okay, thanks for the tip. In tests there are fixture with `.sql` files. Were they generated with [openmaptiles-tools](https://github.com/openmaptiles/openmaptiles-tools#usage-with-docker)?

**Comment by nyurik** [comment]
no, they are simple garden variety postgis sql test files :)

**Comment by berendjan** [comment]
There is a lot to discover trying to build a tile server for maplibre gl js. Like table sources, function sources, raster data, mvt data, mbtiles, pbf files, not to mention hundreds of applications that seem to take different approaches. Is there a forum where i can ask questions to get the what is what and how it’s all connected? Or do you know a good tutorial or starting point?

Also osm2pgsql takes a stylesheet and returns raster data correct? I wanted to be able to change the color in the fly like maputnik. Which was build for mapbox, would this work for osm2pgsql? 
 Or should i use something like imposm3?

**Comment by berendjan** [comment]
for the forum I found gis.stackexchange.com :)

**Comment by berendjan** [comment]
I see, I'm mixing things up already, both osm2pgsql and Imposm just import to a postgis db. Martin then looks up the tables and postgis in the db to create table sources and function sources? am I right?

**Comment by nyurik** [comment]
those utils do create tables, but not functions. OpenMapTiles creates a single function for you though - which you will be able to use from Martin. Currently that function needs to be modified a little bit to match the expected func signature.
